source/FreeRTOS_IP_Utils.c

```diff
@@ -829,10 +829,8 @@ void vPreCheckConfigs( void )
             if( uxSize == 8U )
             {
                 /* This is a 64-bit platform, make sure there is enough space in
-                 * pucEthernetBuffer to store a pointer. */
-                configASSERT( ipconfigBUFFER_PADDING >= 14 );
-                /* But it must have this strange alignment: */
-                configASSERT( ( ( ( ipconfigBUFFER_PADDING ) + 2 ) % 4 ) == 0 );
+                 * pucEthernetBuffer to store a pointer and it must have this strange alignment. */
+                configASSERT( ( ipconfigBUFFER_PADDING >= 14 ) && ( ( ( ( ipconfigBUFFER_PADDING ) + 2 ) % 4 ) == 0 ) );
```

Comment:
It should be ||

Reply:
We need to assert if either `( ipconfigBUFFER_PADDING >= 14 )` or `( ( ( ( ipconfigBUFFER_PADDING ) + 2 ) % 4 ) == 0 )` is false, as `configASSERT` will be triggered if the output of the conditional expression is 0 I think we might need `&&`.

Reply:
Yes if one condition is 0 and other is 1, we should ASSERT, so && make sense.